src/content-scripts/inject/cosmjs-offline-signer.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Coin,
+  encodeSecp256k1Signature,
+  OfflineSigner,
+  serializeSignDoc
+} from "@cosmjs/launchpad";
+import { AccountData, SignResponse } from "@cosmjs/launchpad/types/signer";
+import {
+  GetKeyMsg,
+  RequestSignMsg,
+  RequestTxBuilderConfigMsg
+} from "../../background/keyring";
+import { sendMessage } from "../../common/message/send";
+import { BACKGROUND_PORT } from "../../common/message/constant";
+import { StdSignDoc } from "@cosmjs/launchpad/types/encoding";
+import { feeFromString } from "../../background/keyring/utils";
+
+const Buffer = require("buffer/").Buffer;
+
+export class CosmJSOfflineSigner implements OfflineSigner {
+  constructor(private readonly chainId: string) {}
+
+  async getAccounts(): Promise<AccountData[]> {
+    const key = await sendMessage(BACKGROUND_PORT, new GetKeyMsg(this.chainId));
+
+    return [
+      {
+        address: key.bech32Address,
+        // Currently, only secp256k1 is supported.
+        algo: "secp256k1",
+        pubkey: Buffer.from(key.pubKeyHex)
+      }
+    ];
```

Comment:
Does Keplr support multiple accounts for one chain (`m/44'/118'/0'/0/0`, `m/44'/118'/0'/0/1`, `m/44'/118'/0'/0/2` …)?

Reply:
Keplr supports multiple accounts. Users can make the accounts according to the mnemonic, HD path, **private key**, or ledger... And, it is impossible for the user to change the HD path after the account was made. Currently, the private key type is only used to support the Torus sign-in. To support multiple key types, we made it to return the account that is now selected when returning account information.

Reply:
okay, thank you!